File: structural/java/adapter/src/Triangle.java
public class Triangle implements Shape{
    private double width;
    private double height;
    public Triangle(double width, double height){
        this.width  = width;
        this.height = height;
    }
    @Override
    public void draw() {
        int width = 1;
        for(int i=0;i<this.height; i++){
            for(int j=0; j<(this.height - width); j++) {
                System.out.print(" ");
            }
            for(int j=0; j<width * 2; j++) {
                System.out.print("*");
            }
            width = width + 1;
            System.out.println();
        }
    }

    @Override
    public double getWidth() {
        return this.width;
    }

    @Override
    public double getHeight() {
        return this.height;
    }
}

File: structural/python/adapter/Triangle.py
from Shape import Shape
class Triangle(Shape):
    __width:  float = 0.0
    __height: float = 0.0
    def __init__(self, width, height):
        self.__width  = width
        self.__height = height

    def draw(self) ->None:
        width:int = 1
        for i in range(self.__height):
            for j in range(self.__height - width):
                print(" ", end="")
            for j in range(width * 2):
                print("*", end="")
            width = width + 1
            print()

    def getWidth(self) ->float:
        return self.__width

    def getHeight(self) ->float:
        return self.__height